### PR TITLE
MxToolbox lookup functionality

### PR DESCRIPTION
**Pull request by devsuds**
The MxToolBox API is a RESTful Web Service allowing MxToolbox customers to query the status of their monitors and run lookups (blacklist, smtp, mx, etc.). Generally, programmers use this API to integrate MxToolBox into their products. For example, customers have created a dashboard on their site which shows the real-time status of their monitors.

### CODE REVIEW

File: fn_mxtoolbox/fn_mxtoolbox/LICENSE [hunk 1]
```diff
@@ -0,0 +1 @@
+<<PUT YOUR LICENSE TEXT HERE>>
```
Comment (MarkScherfling) [inline fn_mxtoolbox/fn_mxtoolbox/LICENSE]: see trello for MIT license

File: fn_mxtoolbox/fn_mxtoolbox/components/fn_mxtoolbox.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
```
Comment (MarkScherfling) [inline fn_mxtoolbox/fn_mxtoolbox/components/fn_mxtoolbox.py]: See coding guideliness for IBM Copyright statement

File: fn_mxtoolbox/fn_mxtoolbox/components/fn_mxtoolbox.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_mxtoolbox", {})
+
+    @function("fn_mxtoolbox")
+    def _fn_mxtoolbox_function(self, event, *args, **kwargs):
+        """Function allowing MxToolbox customers to query the status of their monitors and run lookups (blacklist, smtp, mx, etc.)"""
+        try:
+            # Get the function parameters:
```
Comment (MarkScherfling) [inline fn_mxtoolbox/fn_mxtoolbox/components/fn_mxtoolbox.py]: any error checking needed for these fields?

File: fn_mxtoolbox/setup.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from setuptools import setup, find_packages
+
+setup(
+    name='fn_mxtoolbox',
+    version='1.0.0',
+    license='<<insert here>>',
```
Comment (MarkScherfling) [inline fn_mxtoolbox/setup.py]: Use these values:
    url='https://github.com/ibmresilient/resilient-community-apps',
    license='MIT',
    author='IBM Resilient Labs',
    author_email='resil.labs@gmail.com',
    url='https://ibm.biz/resilientcommunity'

File: fn_mxtoolbox/tests/test_fn_mxtoolbox.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+class TestFnMxtoolbox:
+    """ Tests for the fn_mxtoolbox function"""
+
+    def test_function_definition(self):
+        """ Test that the package provides customization_data that defines the function """
+        func = get_function_definition(PACKAGE_NAME, FUNCTION_NAME)
+        assert func is not None
+
+    @pytest.mark.parametrize("mx_command, mx_argument, mx_param1, expected_results", [
```
Comment (MarkScherfling) [inline fn_mxtoolbox/tests/test_fn_mxtoolbox.py]: we'll talk about this. See if it's needed.

File: fn_mxtoolbox/fn_mxtoolbox/components/fn_mxtoolbox.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        self.options = opts.get("fn_mxtoolbox", {})
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_mxtoolbox", {})
+        self._init()
+
+    def _init(self):
+        validate_fields(['api_token','mx_command','mx_argument'], self.options)
```
Comment (MarkScherfling) [inline fn_mxtoolbox/fn_mxtoolbox/components/fn_mxtoolbox.py]: I think this validation will fail. api_token is correct as it comes from the app.config file, but mx_command and mx_argument will come through the function invocation from resilient.

**Comment by hpyle** [comment]
Merging per discussion with Mark

**Pull request merged**